XLS-39d-clawback/README.md

```diff
@@ -0,0 +1,182 @@
+
+
+### 3.3. Transactions
+This proposal introduces one new transaction: `Clawback`
+
+#### 3.3.1. `Clawback` transaction
+The **`Clawback`** transaction modifies a trustline object, by adjusting the balance accordingly and, if instructed to, by changing relevant flags. If possible (i.e. if the `Clawback` transaction would leave the trustline is in the "default" state), the transaction will also remove the trustline.
+
+**An counterparty can be clawed back if and only if the trustline has been frozen (or global freeze is enabled) and `lsfAllowClawback` is set.** There are two ways to enable freeze on a trustline, either through a `TrustSet` or `Clawback` where the `Flags` field is set accordingly. If a clawback transaction is attempted on a trustline that has not been frozen, the transaction will not be allowed and will return with an error code `tecNO_PERMISSION`.
+
+The transaction supports all the existing "common" fields for a transaction.
+
+##### Transaction-specific Fields
+
+| Field Name          | Required?        |  JSON Type    | Internal Type     |
+|---------------------|:----------------:|:-------------:|:-----------------:|
+| `TransactionType`   |:heavy_check_mark:|`string`       |   `UINT16`        |
+
+Indicates the new transaction type **`Clawback`**. The integer value is `30`. The recommended name is `ttCLAWBACK`.
+
+| Field Name    | Required?        |  JSON Type  | Internal Type     |
+|---------------|:----------------:|:-----------:|:-----------------:|
+| `Account`     |:heavy_check_mark:|`string`     | `ACCOUNT ID`      |
+
+Indicates the account which is executing this transaction. The account **MUST** be the issuer of the asset being clawed back. Note that in the XRP Ledger, trustlines are bidirectional and, under some configurations, both sides can be seen as the "issuer" of an asset. In this specification, the term issuer is used to mean the side of the trustline that has an outstanding balance (i.e. 'owes' the issued asset) that it wishes to claw back.
+
+---
+
+| Field Name | Required?        | JSON Type | Internal Type |
+|------------|:----------------:|:---------:|:-------------:|
+| `Amount`   |:heavy_check_mark:| `object`  |   `AMOUNT`    |
+
+Indicates the amount being clawed back, as well as the counterparty from which the amount is being clawed back from. It is not an error if the amount exceeds the holder's balance; in that case, the maximum available balance is clawed back. It is not an error if the amount is zero; in this case, the transaction does not claw back any funds but may adjust flags, as indicated by the `Flags` field.
+
+It is an error if the counterparty listed in `Amount` is the same as the `Account` issuing this transaction; the transaction should fail execution with `temBAD_AMOUNT`.
```

Comment:
The example in 3.3.2 also indicates that it is an error if no trustline exists, or the trustline is set to 0 (does this mean the balance is 0?). I think it would make more sense to have that specified here along with the other possible errors, instead of as part of an example.

Reply:
changed